src/Controller/Api/DatasetSubmissionController.php

```diff
@@ -392,6 +396,44 @@ public function getFiles(int $id)
         return $fileData;
     }
 
+    /**
+     * Adds a file to a dataset submission.
+     *
+     * @param DatasetSubmission      $datasetSubmission The id of the dataset submission.
+     * @param Request                $request           The request body sent with file metadata.
+     * @param EntityManagerInterface $entityManager     Entity manager interface to doctrine operations.
+     *
+     * @Route(
+     *     "/api/files_dataset_submission/{id}",
+     *     name="pelagos_api_add_file_dataset_submission",
+     *     methods={"POST"},
+     *     defaults={"_format"="json"},
+     *     requirements={"id"="\d+"}
+     *     )
+     *
+     * @View()
+     *
+     * @return Response
+     */
+    public function addFile(DatasetSubmission $datasetSubmission, Request $request, EntityManagerInterface $entityManager)
+    {
+        $this->denyAccessUnlessGranted('IS_AUTHENTICATED_FULLY');
```

Comment:
If file already exists in filelist, mark other one as FILE_DELETE (unless it's still FILE_NEW, iow, it's not FILE_DONE or FILE_...).